google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py

```diff
@@ -164,6 +196,26 @@ def dispatch_callback(self, items: Sequence[RequestItem]) -> None:
         if drop_requests:
             self.drop(drop_requests)
 
+    def _handle_duplicate_request_future(
+        self,
+        exactly_once_delivery_enabled: bool,
+        item: Union[requests.AckRequest, requests.ModAckRequest, requests.NackRequest],
+    ) -> None:
+        _LOGGER.debug(
+            "This is a duplicate %s with the same ack_id: %s.",
+            type(item),
+            item.ack_id,
+        )
+        if exactly_once_delivery_enabled and item.future:
```

Comment:
It'll be a bit easier to read if there were two nested IFs. The other one checks for `item.future` and the inner one for `exactly_once_delivery_enabled`. The comments will also be closer to the bifurcation logic.